R/html_paged.R

```diff
@@ -1,4 +1,83 @@
 paged_table_html <- function(x) {
+  "%||%" <- function(x, y) {
+    if(is.null(x)) y else x
+  }
+
+  big_mark <- function(x, ...) {
+    mark <- if (identical(getOption("OutDec"), ",")) "." else ","
+    formatC(x, big.mark = mark, ...)
+  }
+
+  dim_desc <- function(x) {
+    dim <- dim(x) %||% length(x)
+    format_dim <- vapply(dim, big_mark, character(1))
+    format_dim[is.na(dim)] <- "??"
+    paste0(format_dim, collapse = " \u00d7 ")
+  }
+
+  is_atomic <- function(x) {
+    is.atomic(x) && !is.null(x)
+  }
+
+  is_vector <- function(x) {
+    is_atomic(x) || is.list(x)
+  }
+
+  is_vector_s3 <- function(x) UseMethod("is_vector_s3")
```

Comment:
@kevinushey and @hadley Are we okay to have a package private S3 method here or could that interfere somehow with the tibble exported method of the same name? We could just rename it to is_s3_vector and/or just implement it with a switch/case.